### ISSUE TITLE
Add option for transferring file-for-file instead of tar / stream.

### ISSUE THREAD

**Issue by gordan-bobic** [shatteredsilicon/backoid#13]
Some `rclone` backends don't support streaming of a file of unknown size, so `rclone rcat` requires creation of temporary files in `/tmp/` (or where environment variable `TMPDIR` is set to point. One example of such a back end is `azureblob`.

This is a problem when backing up data that is hundreds of gigabytes in size (or anything that won't fit on `/tmp/` mountpoint).

For this, we need a new configuration option:
```
method = [tar | files | zfs-send]
```
Currently we only support `tar`. There is another ticket here for adding progress for zfs-send, but we should add that method explicitly (and bail of `method != zfs-send` is chosen for a `mountpoint=legacy` dataset.

`files` option needs to be added. It should behave as follows (as an example):

```
rclone mkdir remote:bucket/autosnap_2024-09-20_00:01:01_daily
rclone sync zpool/dataset/.zfs/snapshot/autosnap_2024-09-20_00:01:01_daily remote:bucket/autosnap_2024-09-20_00:01:01_daily
```

Because we don't get benefit of compression with this method, this would ideally be used on a `rclone compress` backend.

**Comment by gordan-bobic** [comment]
This will also need to have a --parallel option, so that many files can be transferred in parallel to eliminate single thread bottleneck.
